apps/wallet/src/ui/app/staking/home/Stake.tsx

```diff
@@ -0,0 +1,221 @@
+import { usePendingDelegation } from '../usePendingDelegation';
+import { ActiveDelegation } from './ActiveDelegation';
+import { DelegationCard, DelegationState } from './DelegationCard';
+import BottomMenuLayout, {
+    Menu,
+    Content,
+} from '_app/shared/bottom-menu-layout';
+import Button from '_app/shared/button';
+import { Card, CardItem } from '_app/shared/card';
+import CoinBalance from '_app/shared/coin-balance';
+import { Text } from '_app/shared/text';
+import {
+    activeDelegationIDsSelector,
+    totalActiveStakedSelector,
+} from '_app/staking/selectors';
+import Alert from '_components/alert';
+import Icon, { SuiIcons } from '_components/icon';
+import Loading from '_components/loading';
+import Overlay from '_components/overlay';
+import { useAppSelector, useObjectsState } from '_hooks';
+import { GAS_TYPE_ARG } from '_redux/slices/sui-objects/Coin';
+
+function StakeHome() {
+    const { loading, error, showError } = useObjectsState();
+
+    const activeDelegationIDs = useAppSelector(activeDelegationIDsSelector);
+
+    const [pendingDelegations, { isLoading: pendingDelegationsLoading }] =
+        usePendingDelegation();
+
+    const hasDelegations =
+        activeDelegationIDs.length > 0 || pendingDelegations.length > 0;
+    const [showModal, setShowModal] = useState(true);
+
+    const navigate = useNavigate();
+    const close = useCallback(() => {
+        navigate('/tokens');
+    }, [navigate]);
+
+    const pageTitle = hasDelegations
+        ? 'Stake & Earn SUI'
+        : 'Select a Validator';
+
+    return (
+        <Overlay
+            showModal={showModal}
+            setShowModal={setShowModal}
+            title={!loading && !pendingDelegationsLoading && pageTitle}
+            closeIcon={SuiIcons.Close}
+            closeOverlay={close}
+        >
+            <div className="w-full flex flex-col flex-nowrap h-full overflow-x-scroll">
+                {showError && error ? (
+                    <Alert className="mb-2">
+                        <strong>Sync error (data might be outdated).</strong>{' '}
+                        <small>{error.message}</small>
+                    </Alert>
+                ) : null}
+                <Loading
+                    loading={loading || pendingDelegationsLoading}
+                    className="flex h-full items-center justify-center"
+                >
+                    {hasDelegations && (
+                        <StakedHomeCard
+                            activeDelegationIDs={activeDelegationIDs}
+                            pendingDelegations={pendingDelegations}
+                        />
+                    )}
+                </Loading>
+            </div>
+        </Overlay>
+    );
+}
+
+type StakedCardProps = {
+    activeDelegationIDs: string[];
+    pendingDelegations: {
+        name: string;
+        staked: bigint;
+        validatorAddress: string;
+    }[];
+};
+
+function StakedHomeCard({
```

Comment:
I don't really get why you have this component separation, doesn't seem necessary. Not blocking for merge but confused

Reply:
{hasDelegations ? (
                        <StakedCard
                            activeDelegationIDs={activeDelegationIDs}
                            pendingDelegations={pendingDelegations}
                        />
                    ) : (
                        <ActiveValidatorsCard />
                    )}

Reply:
Is that from a future PR?

Reply:
yes https://github.com/MystenLabs/sui/pull/6856